src/utils/text-mapper.ts

```diff
@@ -0,0 +1,157 @@
+import {
+    ArenaHovedmalKode,
+    OppfolgingsstatusData,
+    ArenaServicegruppeKode
+} from '../data/api/datatyper/oppfolgingsstatus';
+import { OrNothing, StringOrNothing, isNullOrUndefined } from './felles-typer';
+import { PersonaliaV2Info } from '../data/api/datatyper/personalia';
+import { VeilederData } from '../data/api/datatyper/veileder';
+import { TilrettelagtKommunikasjonData } from '../data/api/datatyper/tilrettelagtKommunikasjon';
+
+export function mapServicegruppeTilTekst(servicegruppe: OrNothing<ArenaServicegruppeKode>): string {
+    switch (servicegruppe) {
+        case 'IVURD':
+            return 'Ikke vurdert';
+        case 'OPPFI':
+            return 'Helserelatert arbeidsrettet oppfølging i NAV';
+        case 'VURDI':
+            return 'Sykmeldt, oppfølging på arbeidsplassen';
+        case 'VURDU':
+            return 'Sykmeldt uten arbeidsgiver';
+        case 'BKART':
+            return 'Behov for arbeidsevnevurdering';
+        case 'KAP11':
+            return 'Rettigheter etter Ftrl. Kapittel 11';
+        default:
+            return EMDASH;
+    }
+}
+
+export function mapHovedmalTilTekst(hovedmal: OrNothing<Hovedmal | ArenaHovedmalKode>): string {
+    switch (hovedmal) {
+        case Hovedmal.BEHOLDE_ARBEID:
+        case 'BEHOLDEA':
+            return 'Beholde arbeid';
+        case Hovedmal.OKE_DELTAKELSE:
+        case 'OKEDELT':
+            return 'Øke deltakelse eller mål om arbeid';
+        case Hovedmal.SKAFFE_ARBEID:
+        case 'SKAFFEA':
+            return 'Skaffe arbeid';
+        default:
+            return EMDASH;
+    }
+}
+
+export function mapInnsatsgruppeTilTekst(innsatsgruppe: OrNothing<Innsatsgruppe | ArenaServicegruppeKode>): string {
+    switch (innsatsgruppe) {
+        case Innsatsgruppe.STANDARD_INNSATS:
+        case 'IKVAL':
+            return 'Standardinnsats';
+        case Innsatsgruppe.SITUASJONSBESTEMT_INNSATS:
+        case 'BFORM':
+            return 'Situasjonsbestemt innsats';
+        case Innsatsgruppe.SPESIELT_TILPASSET_INNSATS:
+        case 'BATT':
+            return 'Spesielt tilpasset innsats';
+        case Innsatsgruppe.GRADERT_VARIG_TILPASSET_INNSATS:
+            return 'Gradert varig tilpasset innsats';
+        case Innsatsgruppe.VARIG_TILPASSET_INNSATS:
+        case 'VARIG':
+            return 'Varig tilpasset innsats';
+        default:
+            return EMDASH;
+    }
+}
+
+export function hentOppfolgingsEnhetTekst(
+    oppfolgingsstatus: OppfolgingsstatusData | null | undefined
+): StringOrNothing {
+    if (!oppfolgingsstatus || !oppfolgingsstatus.oppfolgingsenhet) {
+        return null;
+    }
+
+    const {
+        oppfolgingsenhet: { enhetId, navn }
+    } = oppfolgingsstatus;
+
+    if (!`${enhetId}` && !`${navn}`) {
+        return EMDASH;
+    }
+    if (!`${enhetId}`) {
+        return `${navn}`;
+    }
+
+    if (!`${navn}`) {
+        return `${enhetId}`;
+    }
+
+    return `${enhetId} ${navn}`;
+}
```

Comment:
Dersom enten `oppfolgingsstatus` eller `oppfolgingsstatus.oppfolgingsenhet` er `undefined/null` så gir det vel mest mening å returnere default-verdien `EMDASH` i stedet for å returnere `null`?

Dersom det gjelder så kan denne sjekken slåes sammen med sjekken på `enhetId`/`navn`. For å gjøre dette enklest mulig så tror jeg det er best å unngå å bruke destructuring-syntaksen akkurat her:

```suggestion
export function hentOppfolgingsEnhetTekst(
    oppfolgingsstatus: OppfolgingsstatusData | null | undefined
): StringOrNothing {
    const enhetId = oppfolgingsstatus?.oppfolgingsenhet?.enhetId;
    const navn = oppfolgingsstatus?.oppfolgingsenhet?.navn;
    
    if (!enhetId && !navn) {
        return EMDASH;
    }

    if (!enhetId) {
        return `${navn}`;
    }

    if (!navn) {
        return `${enhetId}`;
    }

    return `${enhetId} ${navn}`;
}
```